sql/expression/get_field.go

```diff
@@ -185,7 +192,7 @@ func SchemaToGetFields(s sql.Schema) []sql.Expression {
 	ret := make([]sql.Expression, len(s))
 
 	for i, col := range s {
-		ret[i] = NewGetFieldWithTable(i, 0, col.Type, col.DatabaseSource, col.Source, col.Name, col.Nullable)
+		ret[i] = NewGetFieldWithTable(i+1, 0, col.Type, col.DatabaseSource, col.Source, col.Name, col.Nullable)
```

Comment:
`SchemasToGetFields` is a weird short-term gap for update joins that I've failed to get rid of. Can add comment for dual table offset. The `dual` table generally is definitely a problem. I'm wondering if it can just be flattened with its parent node. Accounting for subquery expressions is probably the tricky part. Subqe, alias references in subqe, dual table columns all kind of collide in unsavory ways. They should be split out into LATERAL joins, eliminating unnecessary dual tables.